tests/unit/linking/test_link.py

```diff
@@ -529,6 +529,115 @@ def test_missing_field_points_exceed_max_missingness_fraction(
         # low-value missing field, but fails user's overall completeness constraint
         assert matches == [False, False]
 
+    def test_no_match_one_suffix_one_not(
+        self,
+        session,
+        default_algorithm,
+        patients: list[schemas.PIIRecord]
+    ):
+        # Make a deep copy of the first patient, give it a suffix
+        patients = [patients[0]]
+        duplicate = copy.deepcopy(patients[0])
+        duplicate.external_id = str(uuid.uuid4())
+        duplicate.name[0].suffix = ["Sr"]
+        patients.append(duplicate)
+
+        matches: list[bool] = []
+        mapped_patients: dict[str, int] = collections.defaultdict(int)
+        for data in patients[:2]:
+            (_, person, results, _) = link.link_record_against_mpi(data, session, default_algorithm)
+            matches.append(bool(person and results))
+            mapped_patients[person.reference_id] += 1
+
+        # First patient inserted into empty MPI, no match
+        # Second patient fails eval in first pass and blocking in second
+        # due to suffix
+        assert matches == [False, False]
```

Comment:
This is the edge case I was interested and was the only hesitation I had with implementing the suffix evaluation for `FIRST_NAME` as opposed to keep it as is and introducing `FULL_FIRST_NAME`. I'll make a follow up ticket up adding Jr/Sr to the edge case appendix with a specific call out for this case.